doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Samsung-Artik-5/index.md

```diff
@@ -5,4 +5,117 @@ permalink: /:path/
+- Botan (only v.1.11) libraries.
+
+To build Avro version higher than 1.7.5, you will need Boost with the iostream library.
+To build Boost with iostreams, you will need to build the zlib and the libbz2 libraries as described below.
+
+Install the third-party components as follows.
+
+1. Install the toolchain.
+
+        wget https://launchpad.net/linaro-toolchain-binaries/trunk/2013.10/+download/gcc-linaro-arm-linux-gnueabihf-4.8-2013.10_linux.tar.bz2
+        tar xvjf gcc-linaro-arm-linux-gnueabihf-4.8-2013.10_linux.tar.bz2
+        sudo mv gcc-linaro-arm-linux-gnueabihf-4.8-2013.10_linux /opt/gcc-linaro-arm-linux-gnueabihf
+
+1. Install zlib (for Avro version higher than 1.7.5).
+
+        wget http://zlib.net/zlib-1.2.8.tar.gz
+        tar -xvzf zlib-1.2.8.tar.gz
+        export INSTALLDIR=/opt/gcc-linaro-arm-linux-gnueabihf
+        export PATH=$INSTALLDIR/bin:$PATH
+        export TARGETMACH=arm-linux-gnueabi
+        export BUILDMACH=i686-pc-linux-gnu
+        export CROSS=arm-linux-gnueabihf
+        export CC=${CROSS}-gcc
+        export LD=${CROSS}-ld
+        export AS=${CROSS}-as
+        cd zlib-1.2.8
+        ./configure --prefix=/opt/gcc-linaro-arm-linux-gnueabihf/arm-linux-gnueabihf/libc/usr
+        make install
+
+1. Install bzip2 (for Avro version higher than 1.7.5).
+
+        wget http://www.bzip.org/1.0.6/bzip2-1.0.6.tar.gz
+        tar -xvzf bzip2-1.0.6.tar.gz
+        export INSTALLDIR=/opt/gcc-linaro-arm-linux-gnueabihf
+        export PATH=$INSTALLDIR/bin:$PATH
+        export TARGETMACH=arm-linux-gnueabi
+        export BUILDMACH=i686-pc-linux-gnu
+        export CROSS=arm-linux-gnueabihf
+        export CC=${CROSS}-gcc
+        export LD=${CROSS}-ld
+        export AS=${CROSS}-as
+        export AR=${CROSS}-ar
+        cd bzip2-1.0.6
+        sed -e "/^all:/s/ test//" Makefile > Makefile-libbz2_so
+        make -f Makefile-libbz2_so CC="${CC}" AR="${AR}"
+        make PREFIX=${INSTALLDIR}/arm-linux-gnueabihf/libc/usr install
+
+
+1. Install Boost (1.54 or higher).
+
+        wget -O boost_1_59_0.tar.bz2 http://sourceforge.net/projects/boost/files/boost/1.59.0/boost_1_59_0.tar.bz2/download
+        tar xvjf boost_1_59_0.tar.bz2
+        cd boost_1_59_0
+        export PATH=$PATH:/opt/gcc-linaro-arm-linux-gnueabihf/bin
+        ./bootstrap.sh
+         sed -r 's/(using\s+gcc)(\s+;)/\1 : arm : arm-linux-gnueabihf-c++\2/g' -i project-config.jam
+         ./bjam install toolset=gcc-arm --prefix=${INSTALLDIR}
+
+1. Install Avro 1.7.5.
+
+        wget https://archive.apache.org/dist/avro/avro-1.7.5/cpp/avro-cpp-1.7.5.tar.gz
+        cd avro-cpp-1.7.5
+        wget https://raw.githubusercontent.com/kaaproject/kaa/8e65212dfb855363e1a8977d4053041c80d785c7/client/client-multi/client-c/toolchains/rpi.cmake
+        wget https://raw.githubusercontent.com/kaaproject/kaa/967970ec57fb0a62c23ffe573385bf0d0299d977/client/client-multi/client-cpp/tools/avro-cpp-disable-tests.patch
+        patch < avro-cpp-disable-tests.patch
```

Comment:
Unfortunately, it is just a hack that removes some code. So it is not yet usable for contributing.